### PR TITLE
Add fields to values schema

### PR DESCRIPTION
**Pull request by ostromart**
This updates missing fields coming from multicluster. 
@ericvn @ayj @elfinhe PTAL if this is everything we need.

### CODE REVIEW

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 1]
```diff
@@ -502,7 +502,15 @@ message GlobalConfig {
 
   google.protobuf.BoolValue operatorManageWebhooks = 41;
 
-  // The next available key is 44
+  // Settings for remote cluster.
+  google.protobuf.BoolValue istioRemote = 44;
```
Comment (elfinhe) [inline pkg/apis/istio/v1alpha1/values_types.proto]: Add comment:
``` go
  // Controls whether to use the Istio remote control plane
  google.protobuf.BoolValue istioRemote = 44;
```

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 2]
```diff
@@ -1028,6 +1036,8 @@ message PilotConfig {
   //
   // See: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
   TypeMapStringInterface podAnnotations = 30;
+
+  PilotConfigSource configSource = 31;
```
Comment (elfinhe) [inline pkg/apis/istio/v1alpha1/values_types.proto]:
```go

  // ConfigSource describes a source of configuration data for networking
  // rules, and other Istio configuration artifacts. Multiple data sources
  // can be configured for a single control plane.
  ConfigSource configSource = 31;
```

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 3]
```diff
@@ -1066,6 +1076,10 @@ message PilotTelemetryConfig {
   google.protobuf.BoolValue enabled = 1;
 }
 
+message PilotConfigSource {
+  string subscribedResources = 1;
+}
+
```
Comment (elfinhe) [inline pkg/apis/istio/v1alpha1/values_types.proto]:
```go

// PilotConfigSource describes information about a configuration store inside a
// mesh. A single control plane instance can interact with one or more data
// sources.
message PilotConfigSource {
  // Describes the source of configuration, if nothing is specified default is MCP
  string subscribedResources = 1;
}
```

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 4]
```diff
@@ -502,7 +502,15 @@ message GlobalConfig {
...
   google.protobuf.BoolValue operatorManageWebhooks = 41;
 
-  // The next available key is 44
+  // Settings for remote cluster.
+  google.protobuf.BoolValue istioRemote = 44;
+  google.protobuf.BoolValue createRemoteSvcEndpoints = 45;
+  google.protobuf.BoolValue remotePilotCreateSvcEndpoint = 46;
+  string remotePolicyAddress = 47;
+  string remotePilotAddress = 48;
+  string remoteTelemetryAddress = 49;
```
Comment (elfinhe) [inline pkg/apis/istio/v1alpha1/values_types.proto]:
```go

  // Specifies the Istio control plane’s pilot Pod IP address or remote cluster DNS resolvable hostname
  string remotePilotAddress = 47;

  // Specifies the Istio control plane’s policy Pod IP address or remote cluster DNS resolvable hostname
  string remotePolicyAddress = 48;

  // Specifies the Istio control plane’s telemetry Pod IP address or remote cluster DNS resolvable hostname
  string remoteTelemetryAddress = 49;
```

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 5]
```diff
@@ -502,7 +502,15 @@ message GlobalConfig {
 
   google.protobuf.BoolValue operatorManageWebhooks = 41;
 
-  // The next available key is 44
+  // Settings for remote cluster.
+  google.protobuf.BoolValue istioRemote = 44;
+  google.protobuf.BoolValue createRemoteSvcEndpoints = 45;
+  google.protobuf.BoolValue remotePilotCreateSvcEndpoint = 46;
```
Comment (elfinhe) [inline pkg/apis/istio/v1alpha1/values_types.proto]: // If set, a selector-less service and endpoint for istio-pilot are created with the remotePilotAddress IP,
  // which ensures the istio-pilot.<namespace> is DNS resolvable in the remote cluster.
  google.protobuf.BoolValue remotePilotCreateSvcEndpoint = 46;

**Review by elfinhe — COMMENTED** [review]
Add some comments?

**Comment by ostromart** [comment]
Thanks for the suggestions, @elfinhe.

**Review by elfinhe — APPROVED** [review]
checked all usages in https://istio.io/docs/setup/install/multicluster/
this is all we need.

**Review by morvencao — APPROVED** [review]

**Comment by howardjohn** [comment]
@ostromart this has a do not merge label

**Pull request merged**